src/pages-iterable.mixin.ts

```diff
@@ -0,0 +1,33 @@
+import { Item } from "./item";
+
+export interface IteratorResult<T> {
+    done: boolean;
+    value: T | null;
+}
+
+export const PagesIterableMixin = (superClass: any) => class extends superClass {
+    public pointer = 0;
+
+    [Symbol.iterator]() {
+        let pointer = 0;
+        this.pointer = pointer
+        const ctx = this;
+
+        return {
+            next(): IteratorResult<Item> {
+                if (pointer < ctx.pages.pages.length) {
+                    return {
+                        done: false,
+                        // @ts-ignore
+                        value: ctx.toString(ctx.pages.pages[pointer++])
```

Comment:
This is a good implementation, the only thing, probably a consumer would like to manipulate with Book/Comics/Magazine object, but in this case consumer receives just a string.

Reply:
In order to achieve this you could use setter/getter to set/get, for example, a page number while iterating.